src/infrastructure/tools.repository.ts

```diff
@@ -0,0 +1,75 @@
+import type NotesApiTransport from './transport/notes-api';
+import Repository from './repository';
+import type EditorTool from '@/domain/entities/EditorTool';
+import type { EditorConfigTool, EditorConfigTools } from '@/domain/entities/EditorTool';
+import type { ToolsStore, ToolsStoreData } from '@/infrastructure/storage/tools';
+import type ToolsRepositoryInterface from '@/domain/tools.repository.interface';
+import { loadScript } from '@/infrastructure/utils/load-script';
+
+/**
+ * Facade for editor tools
+ */
+export default class ToolsRepository
+  extends Repository<ToolsStore, ToolsStoreData>
+  implements ToolsRepositoryInterface
+{
+  /**
+   * Transport instance
+   */
+  private readonly transport: NotesApiTransport;
+
+  /**
+   * Repository constructor
+   *
+   * @param store - stores user data
+   * @param notesApiTransport - notes api transport instance
+   */
+  constructor(store: ToolsStore, notesApiTransport: NotesApiTransport) {
+    super(store);
+
+    this.transport = notesApiTransport;
+  }
+
+  /**
+   * Load or download editor tools
+   *
+   * @param toolsList - list requested tools
+   */
+  public async getTools(toolsList: EditorTool[]): Promise<EditorConfigTools> {
+    const configTools: EditorConfigTools = {};
+
+    for (const tool of toolsList) {
+      const storedTool = this.store.getEditorTool(tool.name);
+
+      if (storedTool) {
+        configTools[tool.name] = storedTool;
+      } else {
+        const downloadedTool = await this.downloadTool(tool);
+
+        if (downloadedTool === undefined) {
+          continue;
+        }
+
+        this.store.setEditorTool(tool.name, downloadedTool);
+        configTools[tool.name] = downloadedTool;
+      }
+    }
+
+    return configTools;
+  }
+
+  /**
+   * Download the user tool
+   *
+   * @param tool - tool for download
+   */
+  private async downloadTool(tool: EditorTool): Promise<EditorConfigTool | undefined> {
```

Comment:
lets move it to the transport level: `tools.transport.ts`

Reply:
I did everything except this comment, i can fix it later. If it needs earlier, please keep this